src/context/FormEventActions/FormEventActions.ts

```diff
@@ -0,0 +1,27 @@
+import { IFormEventActions } from "./IFormEventActions";
+import { CrmSaveMode } from "./CrmSaveMode";
+
+export class FormEventActions implements IFormEventActions {
+
+    constructor(public xrmUi: Xrm.Ui, public xrmEntity: Xrm.Entity, public getAttributeFn: (attributeName: string) => Xrm.Attributes.Attribute) { }

```

Comment:
multiple lines? This is hard to read.

Why are these declared as public? I would make them private and add getters with friendlier names (ex: `ui` instead of `xrmUi`) if you really have to access them.